authentication/src/main/java/life/qbic/authentication/application/user/policy/PasswordResetLinkSupplier.java

```diff
@@ -29,22 +31,32 @@ public PasswordResetLinkSupplier(
       @Value("${service.host.protocol}") String protocol,
       @Value("${service.host.name}") String host,
       @Value("${service.host.port}") int port,
+      @Value("${service.host.proxy-path}") String proxyPath,
       @Value("${password-reset-endpoint}") String resetEndpoint,
       @Value("${password-reset-parameter}") String passwordResetParameter) {
     this.protocol = protocol;
     this.host = host;
     this.port = port;
+    this.proxyPath = proxyPath;
     this.resetEndpoint = resetEndpoint;
     this.passwordResetParameter = passwordResetParameter;
   }
 
   public String passwordResetUrl(String userId) {
     String pathWithQuery = "/" + resetEndpoint + "?" + passwordResetParameter + "=" + userId;
     try {
-      URL url = new URL(protocol, host, port, pathWithQuery);
+      URL url = proxyPath.isBlank() ? urlWithoutProxy(protocol, host, port, pathWithQuery) : urlWithProxy(protocol, host, port, proxyPath, pathWithQuery);
```

Comment:
If you use the servlet properties environment variable you can avoid this dedicated check (since the servlet context will only be set outside of localhost)